container/options.go

```diff
@@ -38,25 +38,49 @@ func applyOptions(c *Container, opts ...Option) error {
 	return nil
 }
 
+// ensure all the container identifiers are either empty or unique.
+func validateIds(c *Container, seen *(map[string]bool)) error {
```

Comment:
Do we need to accept a pointer to the map here?

Maps are actually reference types in Go, they are like a pointer by themselves so this should just work. Here is a generic advice on when to use a pointer:

https://github.com/golang/go/wiki/CodeReviewComments#receiver-type

Reply:
Thanks for the link, very helpful.
Will fix.